src/Components/WebAssembly/WebAssembly/src/Prerendering/ClientComponentParameterDeserializer.cs

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Text.Json;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+
+namespace Microsoft.AspNetCore.Components
+{
+    internal class ClientComponentParameterDeserializer
+    {
+        private readonly ComponentParametersTypeCache _parametersCache;
+
+        public ClientComponentParameterDeserializer(
+            ComponentParametersTypeCache parametersCache)
+        {
+            _parametersCache = parametersCache;
+        }
+
+        public static ClientComponentParameterDeserializer Instance { get; } = new ClientComponentParameterDeserializer(new ComponentParametersTypeCache());
+
+        public ParameterView DeserializeParameters(IList<ComponentParameter> parametersDefinitions, IList<object> parameterValues)
+        {
+            var parametersDictionary = new Dictionary<string, object>(StringComparer.Ordinal);
```

Comment:
Unfortunately, parameter names are case-insensitive.

Reply:
Regarding the use of the dictionary more generally, it'll be fine as long as we're only talking about a few tens or hundred entries. If there are scenarios where people are passing thousands of parameters, we should consider not using a dictionary here, and instead allocating a `RenderTreeFrame[parameters.Count + 1]` and populating that directly just like the internals of `ParameterView.FromDictionary` do.

Personally I prefer the dictionary for clarity as long as we don't expect there to be thousands of parameters in total, because we have to hash all the string keys.

Reply:
I don't expect this will be the case, we would tell people to group the parameters to their root components or create wrapping components in that scenario.

Reply:
done